engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java

```diff
@@ -546,8 +546,8 @@ UnexpectedTypeException getNoValidatorFoundForTypeException(String constraintTyp
 	ValidationException getValidateOnExecutionOnOverriddenOrInterfaceMethodException(Method m);
 
 	@Message(id = 167,
-			value = "A given constraint definition can only be overridden in one mapping file. %1$s is overridden in multiple files")
-	ValidationException getOverridingConstraintDefinitionsInMultipleMappingFilesException(String constraintClass);
+			value = "A given constraint definition can only be overridden once (either through a XML file or the programmatic API). %1$s is overridden in multiple XML files and/or API calls")
```

Comment:
> I think that deviates a bit from the behavior of the DSL in other places.
I wanted to be consistent with the current behavior of the XMLParser: if two definitions of the same constraint are found, even in different files, an exception is thrown.
Now, if the purpose of the programmatic API is not only to offer an *alternative* to the XML, but also a way to *override* the XML, indeed it makes sense to provide a way to ignore the XML.

> So should we similarly add programmatically given definitions, unless includeExistingValidators() is called?
I guess you mean `includeExistingValidators(false)`. I have mixed feelings about that. That would be the most straightforward solution, but that would also be a move from the current use of `includeExistingValidators` in the XML configuration: there, multiple constraint definitions for the same annotation are forbidden, so "existing validators" strongly implies "default validators".
On the other hand, as you suggest, we already have bad names taken from XML configuration that have a broader meaning in the programmatic API.
Also, allowing the user to pick what validators it want to preserve (default, XML, default and XML, or none) would result in major changes in the way we handle validator registration (currently the source of a particular validator registration is not recorded).

So... well, I guess you're right. I can't think of any solution that would be both clearer to the user, consistent with the current programmatic API, consistent with the XML configuration, and would not result in an oversized implementation (compared to what it provides).

I will do as you proposed. On the bright side, I guess the whole thing with changing the `ConstraintHelper` and how I did it will not matter anymore, since I will have to revert it.